src/d3a/models/market/balancing.py

```diff
@@ -26,7 +26,7 @@
 from d3a.models.market.market_structures import BalancingOffer, BalancingTrade
 from d3a.d3a_core.exceptions import InvalidOffer, MarketReadOnlyException, \
     OfferNotFoundException, InvalidBalancingTradeException, \
-    DeviceNotInRegistryError, ChainTradeException
+    DeviceNotInRegistryError
```

Comment:
https://github.com/gridsingularity/d3a/pull/438/files?file-filters%5B%5D=.py#diff-49ce5f7749215295a7b29d8030567663R49

I guess the balancing_offer should also consider the original rate

Reply:
Thanks, I totally forgot about the balancing market. I will adapt it to match the one sided market case.